packages/spl-token/src/domain/worker.ts

```diff
@@ -0,0 +1,263 @@
+    return
+  }
+
+  async onNewAccount(
+    config: AccountIndexerConfigWithMeta<SPLTokenAccount>,
+  ): Promise<void> {
+    const { account, meta } = config
+
+    if (
+      meta.type === SPLTokenType.Mint ||
+      meta.type === SPLTokenType.Account ||
+      meta.type === SPLTokenType.AccountMint
+    ) {
+      const mint = meta.mint
+      if (!this.mints[mint]) {
+        this.mints[mint] = new Mint(
+          mint,
+          this.eventDAL,
+          this.balanceStateDAL,
+          this.balanceHistoryDAL,
+          this.accountMintDAL,
+        )
+      }
+      this.mints[mint].addAccount(account)
+    }
+    console.log('Account indexing', this.context.instanceName, account)
+  }
+
+  async getTimeSeriesStats(
+    account: string,
+    type: string,
+    filters: AccountStatsFilters,
+  ): Promise<any> {
+    return {}
+  }
+
+  async getStats(account: string): Promise<any> {
+    return {}
+  }
+
+  async updateStats(account: string, now: number): Promise<void> {
+    console.log('', account)
+  }
+
+  async getMintEvents(
+    account: string,
+    filters: MintEventsFilters,
+  ): Promise<SPLTokenEvent[]> {
+    const mint = this.mints[account]
+    if (!mint) return []
+    return await mint.getEvents(filters)
+  }
+
+  async getTokenHolders(account: string): Promise<SPLAccountBalance[]> {
+    const mint = this.mints[account]
+    if (!mint) return []
+    return await mint.getTokenHolders()
+  }
+
+  async getAccountHoldings(
+    account: string,
+    filters: AccountHoldingsFilters,
+  ): Promise<SPLAccountHoldings[]> {
+    const mint = this.mints[account]
+    if (!mint) return []
+    return await mint.getTokenHoldings(filters)
+  }
+
+  protected async filterInstructions(
+    ixsContext: InstructionContextV1[],
+  ): Promise<InstructionContextV1[]> {
+    return ixsContext.filter(({ ix }) => isParsedIx(ix))
+  }
+
+  protected async indexInstructions(
+    ixsContext: InstructionContextV1[],
+  ): Promise<void> {
+    const parsedEvents: SPLTokenEvent[] = []
+    const works: PendingWork<MintAccount>[] = []
+    const promises = ixsContext.map(async (ix) => {
+      const account = ix.txContext.parserContext.account
+      if (this.mints[account]) {
+        const parsedIx = this.mintParser.parse(ix, account)
+        if (parsedIx) {
+          if (parsedIx.type === SPLTokenEventType.InitializeAccount) {
+            const work = {
+              id: parsedIx.account,
+              time: Date.now(),
+              payload: {
+                mint: account,
+                timestamp: parsedIx.timestamp,
+                event: parsedIx,
+              },
+            }
+            works.push(work)
+          }
+          if (parsedIx.type === SPLTokenEventType.CloseAccount) {
+            const work = await this.fetchMintDAL.getFirstValueFromTo(
+              [parsedIx.account],
+              [parsedIx.account],
+            )
+            if (work && parsedIx.timestamp >= work.payload.timestamp) {
```

Comment:
We need to check here not only if we have finish from fetching the most recent info (close event), but also if all the history has been fetched (`fetcher.isComplete('backward')`) because the fetching pagination goes from the most recent till the very beginning, so a close event can be fetch on the first page without completing the history fetching